src/urllib3/_collections.py

```diff
@@ -155,7 +155,7 @@ def keys(self) -> set[_KT]:  # type: ignore[override]
             return set(self._container.keys())
 
 
-class HTTPHeaderDictItemView(typing.Set[typing.Tuple[str, str]]):
+class HTTPHeaderDictItemView(typing.Set[typing.Tuple[str, typing.Union[str, bytes]]]):
```

Comment:
What error do you get when using `str | bytes`? Union types are supported by Python 3.10+ (https://docs.python.org/3/library/stdtypes.html#types-union), and I can see other places (line 240) where they appear to work just fine.